edi_purchase_oca/models/purchase_order.py

```diff
@@ -19,3 +19,8 @@ def button_cancel(self):
         if self:
             self._event("on_button_cancel_purchase_order").notify(self)
         return result
+
+    def write(self, vals):
+        if self:
+            self._event("on_button_edit_purchase_order").notify(self, vals)
```

Comment:
better `on_write_purchase_order` to keep the same structure as the other

Reply:
Perfect! Thanks :)